tests/test_snakefmt.py

```diff
@@ -128,6 +132,226 @@ def test_config_adherence_for_code_inside_rules(self, cli_runner, tmp_path):
 
         assert actual.output == expected_output
 
+    def test_check_file_needs_no_changes_exit_code_0(self, cli_runner):
+        stdin = 'include: "a"\n'
+        params = ["--check", "-"]
+
+        actual = cli_runner.invoke(main, params, input=stdin)
+        expected = CheckExitCode.NO_CHANGE.value
+
+        assert actual.exit_code == expected
+
+    def test_check_file_needs_changes_exit_code_1(self, cli_runner):
+        stdin = 'include:  "a"\n'
+        params = ["--check", "-"]
+
+        actual = cli_runner.invoke(main, params, input=stdin)
+        expected = CheckExitCode.WOULD_CHANGE.value
+
+        assert actual.exit_code == expected
+
+    def test_check_file_syntax_invalid_exit_code_123(self, cli_runner):
+        stdin = "foo:  \n"
+        params = ["--check", "-"]
+
+        actual = cli_runner.invoke(main, params, input=stdin)
+        expected = CheckExitCode.INTERNAL_ERROR.value
+
+        assert actual.exit_code == expected
+
+    def test_check_does_not_format_file(self, cli_runner, tmp_path):
+        content = "include: 'a'\nlist_of_lots_of_things = [1, 2, 3, 4, 5, 6, 7, 8]"
+        snakefile = tmp_path / "Snakefile"
+        snakefile.write_text(content)
+        params = ["--check", str(snakefile)]
+
+        result = cli_runner.invoke(main, params)
+
+        expected_exit_code = CheckExitCode.WOULD_CHANGE.value
```

Comment:
any way you can make the exit code only one of {WOULD_CHANGE, NO_CHANGE, PARSING_ERROR}? the `.value` seems redundant as the value is tied to the preceding attribute

Reply:
I've implemented the equality operator for compaing `int` to this enum now.